app/lib/text/generator.rb

```diff
@@ -91,7 +91,7 @@ def components
         @components = @report.active_text_components(@opts)
         @components = components.shuffle
         @nil_priorities, @components = components.partition {|c| c.priority.nil? }
-        @components = components.sort_by {|c| Trigger.priorities[c.priority] }
+        @components = components.sort_by(&:priority_index)
```

Comment:
The rendering is now done in a separate classe. Will look at selecting the components tonigh